gcc/config/riscv/riscv.h

```diff
@@ -513,15 +513,15 @@ enum reg_class
    `crtl->outgoing_args_size'.  */
 #define OUTGOING_REG_PARM_STACK_SPACE(FNTYPE) 1
 
-#define STACK_BOUNDARY 128
+#define STACK_BOUNDARY (TARGET_RVE ? 32 : 128)
```

Comment:
@aswaterman Won't we still need the stack to be 16-byte aligned because of the C.ADDI16SP instruction?  This would change BIGGEST_ALIGNMENT as well.